xcube/core/geom.py

```diff
@@ -722,8 +722,8 @@ def get_dataset_bounds(dataset: Union[xr.Dataset, xr.DataArray],
 
     # Note, x_min > x_max then we intersect with the anti-meridian
     x_bnds_name = get_dataset_bounds_var_name(dataset, x_name)
-    if x_bnds_name:
-        x_bnds_var = dataset.coords[x_bnds_name]
+    if x_bnds_name and x_bnds_name in dataset:
```

Comment:
This seems redundant to me. x_bnds_name is derived from the dataset one line earlier. We know it is in dataset. The fix is the line below. Same comment for y_bnds_name, of course.